src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -817,18 +851,34 @@ static byte[] readInputStreamIntoByteArrayAndClose(InputStream inputStream) thro
     private void copySlaveJarUsingSCP(TaskListener listener, String workingDirectory) throws IOException, InterruptedException {
         SCPClient scp = new SCPClient(connection);
         try {
-            // check if the working directory exists
-            if (connection.exec("test -d " + workingDirectory ,listener.getLogger())!=0) {
+            if(isWindows) {
+              // check if the working directory exists
+              if (connection.exec("Test-Path " + workingDirectory, listener.getLogger()) != 0) {
```

Comment:
Should this add quotation marks around `workingDirectory`? The non-Windows `else` branch doesn't do that either, but directory names with spaces in them might be more common on Windows.

Reply:
Yes, that is probably a good idea